mslib/msui/mpl_qtwidget.py

```diff
@@ -1153,22 +1337,22 @@ class MplTopViewCanvas(MplCanvas):
     def __init__(self, settings=None):
         """
         """
-        super(MplTopViewCanvas, self).__init__()
+        self.myfig = MyTopViewFigure()
+        super(MplTopViewCanvas, self).__init__(self.myfig)
         self.waypoints_interactor = None
         self.satoverpasspatch = []
         self.kmloverlay = None
-        self.map = None
         self.basename = "topview"
 
         # Axes and image object to display the legend graphic, if available.
         self.legax = None
         self.legimg = None
 
         # stores the  topview plot title size(tov_pts) and topview axes label size(tov_als),initially as None.
-        self.tov_pts = None
-        self.tov_als = None
+        self.myfig.tov_pts = None
```

Comment:
this should probably be moved as well. I am just not 100% sure how to best deliver the config_loader setting below to "myfig". I was in a hurry yesterday.